File: codeforces/problemset/490/a/gpt.py
n = int(input())
students = list(map(int, input().split()))

maths = []
programming = []
pe = []

# separate students by their skill
for i in range(n):
  if students[i] == 1:
    programming.append(i+1)
  elif students[i] == 2:
    maths.append(i+1)
  else:
    pe.append(i+1)

num_teams = min(len(maths), len(programming), len(pe))

# output number of teams
print(num_teams)

# output the teams
for i in range(num_teams):
  print(maths[i], programming[i], pe[i])

File: codeforces/problemset/490/a/gpt.go
package main

import (
  "fmt"
)

func main() {
  var n, t int
  fmt.Scan(&n)

  students := make([]int, n)
  maths := []int{}
  programming := []int{}
  pe := []int{}

  // separate students by their skill
  for i := 0; i < n; i++ {
    fmt.Scan(&t)
    students[i] = t
    if t == 1 {
      programming = append(programming, i+1)
    } else if t == 2 {
      maths = append(maths, i+1)
    } else {
      pe = append(pe, i+1)
    }
  }

  num_teams := min(len(maths), len(programming), len(pe))

  // output number of teams
  fmt.Println(num_teams)

  // output the teams
  for i := 0; i < num_teams; i++ {
    fmt.Println(maths[i], programming[i], pe[i])
  }
}

func min(a, b, c int) int {
  if a <= b && a <= c {
    return a
  } else if b <= a && b <= c {
    return b
  } else {
    return c
  }
}

